### PR TITLE
#162379574 Create user groups

### PR DESCRIPTION
**Pull request by Orelongz**
#### What does this PR do?
Allows a user to create an album

#### Description of Task to be completed?
- create user group model and controller
- write tests for model and controller
- implement feedback on tests
- add active serializers
- add internationalization

#### How should this be manually tested?
- Run `rake db:migrate`
- Run `rails s`
- Using postman, send a `POST` request to `localhost:3000/user_groups`

#### What are the relevant pivotal tracker stories?
[#162379574](https://www.pivotaltracker.com/story/show/162379574)

#### Any background context you want to add?
N/A

#### Screenshots
screen shot 2018-12-07 at 10 24 23 am

### CODE REVIEW

File: spec/models/user_group_spec.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require "rails_helper"
+
+RSpec.describe UserGroup, type: :model do
+  describe UserGroup do
```
Comment (sayidu) [inline spec/models/user_group_spec.rb]: We need only one ` describe UserGroup do`.

**Comment by gr1d99** [comment]
good work @Orelongz, please also refer to this https://github.com/andela/bp-andela-media/pull/10#issuecomment-443119394

**Comment by gr1d99** [comment]
Add [Rails Internationalization (I18n)](https://guides.rubyonrails.org/i18n.html)

File: config/application.rb [hunk 2]
```diff
@@ -27,7 +27,10 @@ class Application < Rails::Application
         controller_specs: true,
         request_specs: true
       g.fixture_replacement :factory_bot, dir: "spec/factories"
-  end
+    end
     # config.active_job.queue_adapter = :sidekiq
+    config.generators do |g|
```
Comment (faithngetich) [inline config/application.rb]: What is g? It would be nice to use a descriptive name.

Comment (Orelongz) [reply config/application.rb]: @faithngetich, the `g` is a boiler_plate code and refers to `generator`. I have moved the content of this code into the existing `config.generators` block in the file.

File: app/models/user_group.rb [hunk 3]
```diff
@@ -1,5 +1,11 @@
 class UserGroup < ApplicationRecord
-  validates :name, presence: true,
-                   uniqueness: { message: "already exists" },
-                   length: { maximum: 32 }
+  translates :name
+  validates_associated :translations
+
+  validates :name,
+    presence: true,
+    uniqueness: { message: "already exists" },
```
Comment (gr1d99) [inline app/models/user_group.rb]: Add message to locale files for example

**en.yml**
```yml
en:
  user_groups:
    uniqueness: 'already exists'
```

then you would look it up using `I18n`
**user_group.rb**
```ruby
...
validates :name,
  presence: true,
  uniqueness: { message: I18n.t("user_groups.uniqueness") },
...
```

File: config/initializers/locale.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+I18n.load_path += Dir[Rails.root.join('lib', 'locale', '*.{rb,yml}')]
+ 
+# Whitelist locales available for the application
+I18n.available_locales = [:en, :es]
+ 
+# Set default locale to something other than :en
+# I18n.default_locale = :es
```
Comment (gr1d99) [inline config/initializers/locale.rb]: add a new line

**Pull request merged**